src/Sarif/Core/FileData.cs

```diff
@@ -42,9 +70,37 @@ public static FileData Create(Uri uri, bool computeHashes)
                                 Algorithm = AlgorithmKind.Sha256,
                             },
                         };
+                }
+
             }
+            catch (Exception e) when (e is IOException || e is UnauthorizedAccessException) { }
 
             return fileData;
         }
+
+        private static string EncodeFileContents(string filePath, string mimeType, Encoding inputFileEncoding)
+        {
+            byte[] fileContents;
+
+            if (mimeType != SarifWriters.MimeType.Binary)
```

Comment:
>mimeType != SarifWriters.MimeType.Binary [](start = 16, length = 40)

This is the heart of the problem. We can't reliably detect "text-ness" from MIME type.